ceci/stage.py

```diff
@@ -783,6 +783,68 @@ def split_tasks_by_rank(self, tasks):
             if i % self.size == self.rank:
                 yield task
 
+    def map_tasks_by_rank(self, function, tasks, allgather=False):
+        """Run a function over a series of tasks in parallel
+
+        This mirrors the map function, and returns the equivalent of
+        [function(task) for task in tasks], but executes in parallel.
+
+        Parameters
+        ----------
+        function: Callable
+            Function to be run on tasks
+
+        tasks: Iterable
```

Comment:
It is stated explicit above that function(task) gets called for each task.  But this description makes it sound like "task" is something special, rather that just the data that get passed to the function.

Reply:
Thanks - good point. I've changed this to "inputs" to be more generic